src/tests/require-adddisplayname.coffee

```diff
@@ -0,0 +1,75 @@
+{rules: {'require-adddisplayname': rule}} = require '..'
+{RuleTester} = require 'eslint'
+
+ruleTester = new RuleTester()
+
+error =
+  message: 'Set a display name with addDisplayName()'
+
+tests =
+  valid: [
+    code: '''
+      flowMax()
+    '''
+  ,
+    code: '''
+      const Component = flowMax(
+        addDisplayName('Component'),
+        ({b}) => <div>{b}</div>
+      )
+    '''
+  ,
+    code: '''
+      const Component = flowMax(
+        addDisplayName('Something'),
+        addProps({b: 2}),
+        ({b}) => <div>{b}</div>
+      )
+    '''
+  ,
+    # not definitively a "component"
+    code: '''
+      flowMax(
+        ({b}) => <div>{b}</div>
+      )
+    '''
+  ]
+  invalid: [
+    code: '''
+      const Component = flowMax(
+        ({b}) => <div>{b}</div>
+      )
+    '''
+    settings:
+      'ad-hok/should-fix-flow-flowmax': yes
```

Comment:
The setting name `should-fix-flow-flowmax` is now inaccurate, but effectively it is fine for it to be a single shared setting (because it basically means "are you using known-imports"), so should probably just rename as a breaking change at some point